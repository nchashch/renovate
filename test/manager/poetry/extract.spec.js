const fs = require('fs');
const { extractPackageFile } = require('../../../lib/manager/poetry/extract');

const pyproject1toml = fs.readFileSync(
  'test/datasource/poetry/_fixtures/pyproject.1.toml',
  'utf8'
);

describe('lib/manager/poetry/extract', () => {
  describe('extractPackageFile()', () => {
    let config;
    beforeEach(() => {
      config = {};
    });
    it('returns null for empty', () => {
      expect(extractPackageFile('nothing here', config)).toBe(null);
    });
    it('extracts multiple dependencies', () => {
      const res = extractPackageFile(pyproject1toml, config);
      expect(res.deps).toMatchSnapshot();
      // expect(res.deps).toHaveLength(6);
    });
  });
});
