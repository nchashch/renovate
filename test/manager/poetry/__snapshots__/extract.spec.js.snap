// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/poetry/extract extractPackageFile() extracts multiple dependencies (with dep = {version = "1.2.3"} case) 1`] = `
Array [
  Object {
    "currentValue": "*",
    "datasource": "pypi",
    "depName": "dep1",
  },
  Object {
    "currentValue": "^0.6.0",
    "datasource": "pypi",
    "depName": "dep2",
  },
  Object {
    "currentValue": "^0.33.6",
    "datasource": "pypi",
    "depName": "dep3",
  },
  Object {
    "currentValue": "",
    "datasource": "pypi",
    "depName": "dep4",
    "skipReason": "path-dependency",
  },
  Object {
    "currentValue": "^0.8.3",
    "datasource": "pypi",
    "depName": "extra_dep1",
    "skipReason": "path-dependency",
  },
  Object {
    "currentValue": "^0.9.4",
    "datasource": "pypi",
    "depName": "extra_dep2",
    "skipReason": "path-dependency",
  },
  Object {
    "currentValue": "^0.4.0",
    "datasource": "pypi",
    "depName": "extra_dep3",
    "skipReason": "path-dependency",
  },
]
`;

exports[`lib/manager/poetry/extract extractPackageFile() extracts multiple dependencies 1`] = `
Array [
  Object {
    "currentValue": "*",
    "datasource": "pypi",
    "depName": "dep1",
  },
  Object {
    "currentValue": "^0.6.0",
    "datasource": "pypi",
    "depName": "dep2",
  },
  Object {
    "currentValue": "^0.33.6",
    "datasource": "pypi",
    "depName": "dep3",
  },
  Object {
    "currentValue": "^3.0",
    "datasource": "pypi",
    "depName": "dev_dep1",
  },
  Object {
    "currentValue": "Invalid version.",
    "datasource": "pypi",
    "depName": "dev_dep2",
    "skipReason": "unknown-version",
  },
  Object {
    "currentValue": "^0.8.3",
    "datasource": "pypi",
    "depName": "extra_dep1",
  },
  Object {
    "currentValue": "^0.9.4",
    "datasource": "pypi",
    "depName": "extra_dep2",
  },
  Object {
    "currentValue": "^0.4.0",
    "datasource": "pypi",
    "depName": "extra_dep3",
  },
]
`;

exports[`lib/manager/poetry/extract extractPackageFile() handles case with no dependencies 1`] = `Array []`;

exports[`lib/manager/poetry/extract extractPackageFile() handles multiple constraint dependencies 1`] = `
Array [
  Object {
    "currentValue": "",
    "datasource": "pypi",
    "depName": "foo",
    "skipReason": "multiple-constraint-dep",
  },
]
`;
